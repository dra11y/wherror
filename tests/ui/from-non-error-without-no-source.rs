use wherror::Error;

/// Sounds like bad grammer, but it's not!
/// Test that #[from] NotErrorWithoutNoSource fails
/// when the user should use #[from(no_source)] NotErrorWithoutNoSource
#[derive(Debug)]
struct NotErrorWithoutNoSource;

#[derive(Error, Debug)]
#[error("...")]
pub enum E {
    Bad(#[from] NotErrorWithoutNoSource),
}

fn main() {}
