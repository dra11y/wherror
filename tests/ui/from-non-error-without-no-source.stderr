error[E0599]: the method `as_dyn_error` exists for reference `&NotErrorWithoutNoSource`, but its trait bounds were not satisfied
 --> tests/ui/from-non-error-without-no-source.rs:12:9
  |
 7 | struct NotErrorWithoutNoSource;
   | ------------------------------ doesn't satisfy `NotErrorWithoutNoSource: AsDynError<'_>` or `NotErrorWithoutNoSource: std::error::Error`
...
12 |     Bad(#[from] NotErrorWithoutNoSource),
   |         ^^^^^^^ method cannot be called on `&NotErrorWithoutNoSource` due to unsatisfied trait bounds
   |
   = note: the following trait bounds were not satisfied:
           `NotErrorWithoutNoSource: std::error::Error`
           which is required by `NotErrorWithoutNoSource: AsDynError<'_>`
           `&NotErrorWithoutNoSource: std::error::Error`
           which is required by `&NotErrorWithoutNoSource: AsDynError<'_>`
note: the trait `std::error::Error` must be implemented
  --> $RUST/core/src/error.rs
   |
   | pub trait Error: Debug + Display {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `as_dyn_error`, perhaps you need to implement it:
           candidate #1: `AsDynError`
